### PR TITLE
Create `codeql_scrub.yml` to add scrub post processing to scans

### LINKED ISSUE #169: Configure CodeQL to use SCRUB for external analysis
**Describe the sustaining activity**
Use [SCRUB](https://github.com/nasa/scrub) to format CodeQL security scans for analysis by the MGSS security team.

**Expected behavior**
Output can be consumed for analysis in a consistent format

**Additional context**

### PR DESCRIPTION
**Pull request by davetsay**
Closes #169
see https://nasa.github.io/scrub/

### CODE REVIEW

File: .github/workflows/codeql_scrub.yml [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+#
+# You may wish to alter this file to override the set of languages analyzed,
+# or to provide custom queries or build logic.
+#
+# ******** NOTE ********
+# We have attempted to detect the languages in your repository. Please check
+# the `language` matrix defined below to confirm you have the correct set of
+# supported CodeQL languages.
+#
+name: "CodeQL"
```
Comment (lylebarner) [inline .github/workflows/codeql_scrub.yml:103]: You should be able to delete lines 103-155. This is a duplicate of the steps included in the lines above.

File: .github/workflows/codeql_scrub.yml [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+    - name: Checkout repository
+      uses: actions/checkout@v4
+
+    # Initializes the CodeQL tools for scanning.
+    - name: Initialize CodeQL
+      uses: github/codeql-action/init@v3
+      with:
+        languages: ${{ matrix.language }}
+        build-mode: ${{ matrix.build-mode }}
+        # If you wish to specify custom queries, you can do so here or in a config file.
```
Comment (lylebarner) [inline .github/workflows/codeql_scrub.yml:64]: Add the following line here after the `build-mode` attribute to enable all of the security checks.

```yaml
queries: security-and-quality, security-extended
```

**Review by lylebarner — CHANGES_REQUESTED** [review]
I think if you make the changes outlined in the comments you should be good to go.

File: .github/workflows/codeql_scrub.yml (hunk 1 above)
Comment (davetsay) [reply .github/workflows/codeql_scrub.yml]: oh wow. when i said i copy/pasted your config i really meant it 😅

**Comment by davetsay** [comment]
> I think if you make the changes outlined in the comments you should be good to go.

Thanks for the review. I've made the changes @lylebarner

File: .github/workflows/codeql_scrub.yml [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+    #   - https://gh.io/supported-runners-and-hardware-resources
+    #   - https://gh.io/using-larger-runners
+    # Consider using larger runners for possible analysis time improvements.
+    runs-on: ${{ (matrix.language == 'swift' && 'macos-latest') || 'ubuntu-latest' }}
+    timeout-minutes: ${{ (matrix.language == 'swift' && 120) || 360 }}
+    permissions:
+      # required for all workflows
+      security-events: write
+
+      # only required for workflows in private repositories
```
[the remark this answers is not in the document]
Comment (davetsay) [reply .github/workflows/codeql_scrub.yml:36]: I think you're right in that we don't. I see this in other public workflows, including Open MCT's codeql, so I think we are also okay leaving it in.

**Review by jvigliotta — APPROVED** [review]
LGTM!

**Review by lylebarner — APPROVED** [review]
Looks good! The action seems to be generating the appropriate output files.

**Comment by davetsay** [comment]
> Looks good! The action seems to be generating the appropriate output files.

Thanks so much for your help @lylebarner !

**Pull request merged**